client/python/lib/test/test_tensors.py

```diff
@@ -111,42 +111,63 @@ def test_make_tensor_proto_valid_int():
     np_values = np.array(values, np.int8)
     assert tensor_proto.tensor_content == np_values.tobytes()
     assert tensor_proto.dtype == DataType.DT_INT8
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
+
+def test_make_tensor_proto_valid_empty_list():
+    values = []
+    tensor_proto = make_tensor_proto(shape=[0], dtype = DataType.DT_INT8, values = values)
+    np_values = np.array(values, np.int8)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_INT8
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=0)])
+
+def test_make_tensor_proto_valid_empty_list_of_empty_lists():
+    values = [[],[],[]]
```

Comment:
Here you add test which is similar to the one which could expose endless loop isssue. The only thing that is missing is sestting shape to None. Please add such test so we are sure we will not get hang.